files/KoreBuild/scripts/KoreBuild.psm1

```diff
@@ -15,200 +15,46 @@ function Join-Paths($path, $childPaths) {
 Set-Variable 'IS_WINDOWS' -Scope Script -Option Constant -Value $((Get-Variable -Name IsWindows -ValueOnly -ErrorAction Ignore) -or !(Get-Variable -Name IsCoreClr -ValueOnly -ErrorAction Ignore))
 Set-Variable 'EXE_EXT' -Scope Script -Option Constant -Value $(if ($IS_WINDOWS) { '.exe' } else { '' })
 
-<#
-.SYNOPSIS
-Builds a repository
-
-.DESCRIPTION
-Invokes the default MSBuild lifecycle on a repostory. This will download any required tools.
-
-.PARAMETER Path
-The path to the repository to be compiled
-
-.PARAMETER MSBuildArgs
-Arguments to be passed to the main MSBuild invocation
-
-.EXAMPLE
-Invoke-RepositoryBuild $PSScriptRoot /p:Configuration=Release /t:Verify
-
-.NOTES
-This is the main function used by most repos.
-#>
-function Invoke-RepositoryBuild(
-    [Parameter(Mandatory = $true)]
-    [string] $Path,
-    [Parameter(ValueFromRemainingArguments = $true)]
-    [string[]] $MSBuildArgs) {
-
-    $ErrorActionPreference = 'Stop'
-
-    if (-not $PSBoundParameters.ContainsKey('Verbose')) {
-        $VerbosePreference = $PSCmdlet.GetVariableValue('VerbosePreference')
-    }
+function Set-Settings
+{
+    [CmdletBinding()]
+    param (
+        [string]$ToolsSource,
+        [string]$DotNetHome,
+        [string]$Path,
+        [string]$ConfigFile 
+    )
+    $global:ToolsSource = $ToolsSource
+    $global:DotNetHome = $DotNetHome
+    $global:Path = $Path
+    $global:SDKVersion = __get_dotnet_sdk_version
+}
 
-    $Path = Resolve-Path $Path
-    Push-Location $Path | Out-Null
-    try {
-        Write-Verbose "Building $Path"
-        Write-Verbose "dotnet = ${global:dotnet}"
-
-        # Generate global.json to ensure the repo uses the right SDK version
-        $sdkVersion = __get_dotnet_sdk_version
-        if ($sdkVersion -ne 'latest') {
-            "{ `"sdk`": { `"version`": `"$sdkVersion`" } }" | Out-File (Join-Path $Path 'global.json') -Encoding ascii
-        } else {
-            Write-Verbose "Skipping global.json generation because the `$sdkVersion = $sdkVersion"
+function Invoke-Command(
```

Comment:
You'll need to choose a different name. This one is taken :) 

https://docs.microsoft.com/en-us/powershell/module/Microsoft.PowerShell.Core/Invoke-Command?view=powershell-6